lib/improver/weights.py

```diff
@@ -0,0 +1,251 @@
+
+            Args:
+                cube : iris.cube.Cube
+                       Cube to blend across the coord.
+                coord : string
+                        The name of a coordinate dimension in the cube.
+            Returns:
+                weights : array of weights, sum of all weights = 1.0
+        """
+        if not isinstance(cube, iris.cube.Cube):
+            msg = ('The first argument must be an instance of '
+                   'iris.cube.Cube but is'
+                   ' {0:s}'.format(type(cube)))
+            raise ValueError(msg)
+
+        if not cube.coords(coord):
+            msg = ('The coord for this plugin must be '
+                   'an existing coordinate in the input cube')
+            raise ValueError(msg)
+
+        num_of_weights = len(cube.coord(coord).points)
+
+        weights = linear_weights(num_of_weights, y0val=self.y0val,
+                                 slope=self.slope,
+                                 ynval=self.ynval)
+
+        return weights
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = '<ChooseBasicLinearWeights y0val={0:4.1f}'.format(self.y0val)
+        if self.ynval is None:
+            desc += ', slope={0:6.2f}>'.format(self.slope)
+        else:
+            desc += ', ynval={0:4.1f}>'.format(self.slope)
+        return desc
+
+
+class ChooseDefaultWeightsNonLinear(object):
+    """ Calculate Default Weights using NonLinear Function. """
+    def __init__(self, cval=0.85):
+        """Set up for calculating default weights using lnon-inear function
+            cval = float: value greater than 0, less than equal 1.0
+                   default = 0.85
+                   equal weights when cval = 1.0
+        """
+        self.cval = cval
+
+    def process(self, cube, coord):
+        """Calculated weights for a given cube and coord
+
+            Args:
+                cube : iris.cube.Cube
+                       Cube to blend across the coord.
+                coord : string
+                        The name of a coordinate dimension in the cube.
+            Returns:
+                weights : array of weights, sum of all weights = 1.0
+        """
+        if not isinstance(cube, iris.cube.Cube):
+            msg = ('The first argument must be an instance of '
+                   'iris.cube.Cube but is'
+                   ' {0:s}'.format(type(cube)))
+            raise ValueError(msg)
+        if not cube.coords(coord):
+            msg = ('The coord for this plugin must be '
+                   'an existing coordinate in the input cube')
+            raise ValueError(msg)
+        num_of_weights = len(cube.coord(coord).points)
+
+        weights = nonlinear_weights(num_of_weights, cval=self.cval)
+
+        return weights
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = '<ChooseBasicNonLinearWeights cval={0:4.1f}>'.format(self.cval)
```

Comment:
Yes, sorry I changed the name of the Class and forgot to change it here. Now fixed